test/hash/src/test_hash.d

```diff
@@ -195,6 +196,30 @@ void issue19568()
     size_t h = ((ref S3 s3) pure => hashOf(s3))(s3);
 }
 
+/// Check core.internal.convert.toUbyte in CTFE for arrays works with
+/// reference type elements and doesn't call postblits/dtors.
+void issue19582()
+{
+    import core.internal.convert : toUbyte;
+    final static class C : Object {}
+    enum b1 = (() @nogc nothrow pure @safe { C[10] o; return toUbyte(o[])[0]; })();
+
+    static struct S
+    {
+        int x;
+        @disable this(this);
+        ~this() @nogc nothrow
+        {
+            import core.stdc.stdio : puts;
+            if (x) puts("impure");
+        }
+    }
+    enum b2 = () {
+            S[10] a;
+            return ((const S[] a) @nogc nothrow pure @safe => toUbyte(a))(a);
+        }();
+}
```

Comment:
As far the compiler is concerned, `b1` and `b2` are not used, so there's not need to do any semantic analysis on the values they refer to. You should assert the values of `b1` and `b2` are what you expect them to be.

Reply:
I think the test is right since the test doesn't compile without this PR: https://run.dlang.io/is/v2dXCy